examples/vision/patch_conv_net.py

```diff
@@ -0,0 +1,661 @@
+conv_stem = build_convolutional_stem(dimensions=DIMENSIONS)
+conv_trunk = Trunk(depth=TRUNK_DEPTH, dimensions=DIMENSIONS, ratio=SE_RATIO)
+attention_pooling = AttentionPooling(dimensions=DIMENSIONS, num_classes=NUM_CLASSES)
+
+patch_conv_net = PatchConvNet(
+    stem=conv_stem,
+    trunk=conv_trunk,
+    attention_pooling=attention_pooling,
+    train_augmentation_model=train_augmentation_model,
+    test_augmentation_model=test_augmentation_model,
+)
+
+# Assemble the callbacks.
+train_callbacks = [TrainMonitor(epoch_interval=5)]
+# Get the optimizer.
+optimizer = tfa.optimizers.AdamW(learning_rate=scheduled_lrs, weight_decay=WEIGHT_DECAY)
+# Compile and pretrain the model.
+patch_conv_net.compile(
+    optimizer=optimizer,
+    loss=keras.losses.SparseCategoricalCrossentropy(from_logits=True),
+    metrics=[
+        keras.metrics.SparseCategoricalAccuracy(name="accuracy"),
+        keras.metrics.SparseTopKCategoricalAccuracy(5, name="top-5-accuracy"),
+    ],
+)
+history = patch_conv_net.fit(
+    train_ds, epochs=EPOCHS, validation_data=val_ds, callbacks=train_callbacks,
+)
+
+# Evaluate the model with the test dataset.
+loss, acc_top1, acc_top5 = patch_conv_net.evaluate(test_ds)
+print(f"Loss: {loss:0.2f}")
+print(f"Top 1 test accuracy: {acc_top1*100:0.2f}%")
+print(f"Top 5 test accuracy: {acc_top5*100:0.2f}%")
+
+"""
+## Inference
+
+In this section, we use the trained model to plot the attention map.
+"""
+
+
+def plot_attention(image):
+    """Plots the attention map on top of the image.
+    Args:
+        image: A numpy image of arbitrary size.
+    """
+    # Resize the image to a (32, 32) dim.
+    image = tf.image.resize(image, (32, 32))
+    image = image[tf.newaxis, ...]
+    test_augmented_images = patch_conv_net.test_augmentation_model(image)
+    # Pass through the stem.
+    test_x = patch_conv_net.stem(test_augmented_images)
+    # Pass through the trunk.
+    test_x = patch_conv_net.trunk(test_x)
+    # Pass through the attention pooling block.
+    _, test_viz_weights = patch_conv_net.attention_pooling(test_x)
+    test_viz_weights = test_viz_weights[tf.newaxis, ...]
+    # Reshape the vizualization weights.
+    num_patches = tf.shape(test_viz_weights)[-1]
+    height = width = int(math.sqrt(num_patches))
+    test_viz_weights = layers.Reshape((height, width))(test_viz_weights)
+    selected_image = test_augmented_images[0]
+    selected_weight = test_viz_weights[0]
+    # Plot the images.
+    fig, ax = plt.subplots(nrows=1, ncols=2, figsize=(10, 5))
+    ax[0].imshow(selected_image)
+    ax[0].set_title(f"Original")
+    ax[0].axis("off")
+    img = ax[1].imshow(selected_image)
+    ax[1].imshow(selected_weight, cmap="inferno", alpha=0.6, extent=img.get_extent())
+    ax[1].set_title(f"Attended")
+    ax[1].axis("off")
+    plt.axis("off")
+    plt.show()
+    plt.close()
+
+
+url = "http://farm9.staticflickr.com/8017/7140384795_385b1f48df_z.jpg"
+image = np.asarray(Image.open(requests.get(url, stream=True).raw))
```

Comment:
You can do this without importing `requests` and `Image`, just use `get_file` and `img_to_array`

Reply:
This was a great suggestion! 🚀